bindings/serialize-v8.cc

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "serialize-v8.h"
+#include "serialize.h"
+
+using namespace v8;
+
+class HeapNode : public Node {
+ private:
+  AllocationProfile::Node *node;
+
+ public:
+  HeapNode(AllocationProfile::Node *node) { this->node = node; }
```

Comment:
Can you use initializer lists instead of statements to initialize?

```c++
HeapNode(AllocationProfile::Node* node) : node(node) {}
```

Also, here and elsewhere, use consistent lean for the `*` for pointers. Here it seems to be leaning right. Google style guide wants the `*` to lean left, as does the existing code.

Reply:
Either is fine, the style guide requires to just be consistent - https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions.

Reply:
Switched to list initializer.
The version of clang-format I'm using seems to do right leaning `*` for pointers.

Reply:
> The version of clang-format I'm using seems to do right leaning * for pointers.

I think DerivePointerAlignment is true by default (https://clang.llvm.org/docs/ClangFormatStyleOptions.html) so it just formats everything to the prevalent style of the pointer alignment in the particular source file.

Reply:
Fixed -- looks like clang-format does something clever when trying to decide where to do `*`.